### ISSUE TITLE
  host: Hostname does not match client IP

### ISSUE THREAD

**Issue by pansx** [drawpile/Drawpile#861]
i user kubernetes run docker drawpile,exposed port 27750 to domain dp.yong-gang.cn,people can connect,but when i publish server to drawpile.net it said 

host: Hostname does not match client IP

i think it check is  my pod ip matching my public ip ,can cancel this check?

**Comment by Wade821** [comment]
I don't believe that can be bypassed, probably for security reasons. I'd suggest waiting for @callaa to respond on this. 

But to confirm, you're hosting a server on one address but forwarding the traffic from another address?

**Comment by pansx** [comment]
yes,in kubernetes any docker app will have a pod ip,and using load balancer to expose port,load balancer has public ip and route to pod to access app

**Comment by notfood** [comment]
Append the ` --local-host dp.yong-gang.cn` to the command you use to run drawpile `drawpile-srv`

That will override the kubernetes hostname.

**Comment by pansx** [comment]
i did that at first time but no effect

**Comment by callaa** [comment]
The listing server will resolve the hostname and check that it matches the originating IP address. This is an anti-SPAM measure.

I think most Kubernetes cloud providers will give pods random IP addresses from a pool, so it's not possible to reliably associate them with a specific domain. But I believe it's possible to create an egress service or cloud NAT or something like that to give them a consistent IP. Of course, if it's not the same as the load balancer's IP, there's still a problem.

If this is not feasible, the only solution I can think of is to add support for registered domains + API keys, since I'd rather not just remove the IP check. Without the check, anyone could spam phantom sessions for any domain.

A third, and possibly better, option would be to set up your own list server. The list server can now fetch session listings directly from the server(s) without having to explicitly announce them. (See https://pub.drawpile.net/) Once the new Drawpile communities section is published, this will be the preferred way for servers to list their sessions.

**Comment by pansx** [comment]
default route cant expose port and ingress can only route http...
own list server users wont discover our server unless he setting client and our user good at drawing not good at computer...

**Comment by Wade821** [comment]
@pansx no need to close this yet. Callaa said we have some options, including a new feature possibly....

**Comment by rhwong** [comment]
Same question

This problem occurs when I use [FRP](https://github.com/fatedier/frp) to expose my 27750 port to VPS

Is there a solution now?

**Comment by Wade821** [comment]
The solution just requires time/effort, and there's a variety of other important issues to address. The simplest solution would be to set up your own list server. 

https://github.com/drawpile/listserver
